src/quark.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use translate::*;
+
+#[repr(C)]
+pub struct Quark(ffi::GQuark);
+
+impl Quark {
+    pub fn from_string(s: &str) -> Quark {
+        unsafe {
+            from_glib(ffi::g_quark_from_string(s.to_glib_none().0))
+        }
+    }
+
+    pub fn from_static_string(s: &'static str) -> Quark {
+        unsafe {
+            from_glib(ffi::g_quark_from_static_string(s.to_glib_none().0))
+        }
+    }
+
+    pub fn to_string(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::g_quark_to_string(self.to_glib()))
```

Comment:
The return value has static lifetime and is never NULL. You can directly use CStr here and return a static string slice